src/framework/core/js/Fluid.js

```diff
@@ -808,10 +812,14 @@ var fluid = fluid || fluid_2_0_0;
      * @return The supplied argument, recursively frozen
      */
     fluid.freezeRecursive = function (tofreeze) {
-        fluid.each(tofreeze, function (value) {
-            fluid.freezeRecursive(value);
-        });
-        return fluid.isPlainObject(tofreeze) ? Object.freeze(tofreeze) : tofreeze; // IE11 crashes on freeze of non-object
+        if (fluid.isPlainObject(tofreeze)) {
```

Comment:
Object.freeze() is perfectly well able to freeze arrays - that is, plain ones. It fails to freeze "exotic" kinds of arrays such as those descended from TypedArray. Recall that typeof([]) === "object" in JS ...